libs/partners/anthropic/tests/integration_tests/test_chat_models.py

```diff
@@ -146,6 +146,55 @@ async def test_abatch_tags() -> None:
         assert isinstance(token.content, str)
 
 
+async def test_async_tool_use() -> None:
+    llm = ChatAnthropic(  # type: ignore[call-arg]
+        model=MODEL_NAME,
+    )
+
+    llm_with_tools = llm.bind_tools(
+        [
+            {
+                "name": "get_weather",
+                "description": "Get weather report for a city",
+                "input_schema": {
+                    "type": "object",
+                    "properties": {"location": {"type": "string"}},
+                },
+            }
+        ]
+    )
+    response = await llm_with_tools.ainvoke("what's the weather in san francisco, ca")
+    assert isinstance(response, AIMessage)
+    assert isinstance(response.content, list)
+    assert isinstance(response.tool_calls, list)
+    assert len(response.tool_calls) == 1
+    tool_call = response.tool_calls[0]
+    assert tool_call["name"] == "get_weather"
+    assert isinstance(tool_call["args"], dict)
+    assert "location" in tool_call["args"]
+
+    # Test streaming
+    first = True
+    chunks = []  # type: ignore
+    async for chunk in llm_with_tools.astream(
+        "what's the weather in san francisco, ca"
+    ):
+        chunks = chunks + [chunk]
+        if first:
+            gathered = chunk
+            first = False
+        else:
+            gathered = gathered + chunk  # type: ignore
+    assert len(chunks) > 1
+    assert isinstance(gathered, AIMessageChunk)
+    assert isinstance(gathered.tool_call_chunks, list)
+    assert len(gathered.tool_call_chunks) == 1
```

Comment:
why is this only 1 chunk? Couldn't it be multiple that sum together into a single tool call?

Reply:
The chunks here combine into one tool call